### PR TITLE
enh(ruby) add support for real, imaginary, and e-notation numbers

### LINKED ISSUE #2766: (ruby) numeric literals with suffixes (exponent, real, imaginary) not highligted
**Describe the issue**

In Ruby, float, rational, and complex literals do not highlight the textual portion of the literal. 

```
3.5e4
2r
6i
```

The `e4`, `r`, and `i` remain unhighlighted.

**Which language seems to have the issue?**

Ruby.

**Expected behavior**

The textual portion should also be highlighted as `number`, as the Python grammar and others do.

### PR DESCRIPTION
**Pull request by khamer**
Changes to regex for matching numbers in ruby. Resolves #2766.

### Changes
Changes the number regex for ruby to support exponential notation (1e123) as well as adding a trailing `r` or `i` - for real or imaginary numbers.

### Checklist
- [x] Added new markup/ruby/numbers.txt tests
- [x] Added a note to the changelog
- [x] Added myself to `AUTHORS.txt` under Contributors

### CODE REVIEW

File: src/languages/ruby.js [hunk 1]
```diff
@@ -128,7 +128,7 @@ export default function(hljs) {
     },
     {
       className: 'number',
-      begin: '(\\b0[0-7_]+)|(\\b0x[0-9a-fA-F_]+)|(\\b[1-9][0-9_]*(\\.[0-9_]+)?)|[0_]\\b',
+      begin: '(\\b0[0-7_]+)|(\\b0x[0-9a-fA-F_]+)|(\\b[1-9][0-9_]*(\\.[0-9_]+)?([ri]|[eE][-+]?[0-9_]+i?)?)|[0_]\\b',
```
Comment (joshgoebel) [inline src/languages/ruby.js:131]: Please break these out into variants as in the python?

File: test/markup/ruby/numbers.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+number = 1.0e1_2
+number = 1.0e1_2i
+number = 1.0r
```
Comment (joshgoebel) [inline test/markup/ruby/numbers.txt:3]: A few more tests for `0.` and some other edge cases?

**Review by joshgoebel — CHANGES_REQUESTED** [review]
Please update (see notes).

**Comment by joshgoebel** [comment]
Wait, are you all doing the same things? 

https://github.com/highlightjs/highlight.js/pull/2786
https://github.com/highlightjs/highlight.js/pull/2788

Should one be build on the other or should one of you wait?  Feels like we're duplicating effort here.

**Comment by gibson042** [comment]
This PR is less correct and complete than #2788, which follows the patterns from #2780. I believe it should be closed.

**Comment by joshgoebel** [comment]
Sorry @khamer I think that's accurate and we need to go with #2788 on this one.  There are some other low hanging fruit issues if you'd like to find a different issue to contribute to. Not sure if you were jus looking to help out or really wanted to fix this one particular issue.  Sorry for any confusion.
